A panic occurs when tidb_enable_ordered_result_mode is enabled

## Bug Report
开启 tidb_enable_ordered_result_mode 后，SQL 执行 panic

### 1. Minimal reproduce step (Required)
<img width="473" alt="8a55761aad6af508ff9754fb2dd900c" src="https://github.com/pingcap/tidb/assets/52245161/324ce8c5-ea9f-4b80-9f78-e33263a259d7">


### 2. What did you expect to see? (Required)
执行正常

### 3. What did you see instead (Required)
[2023/06/25 10:18:28.572 +08:00] [ERROR] [conn.go:1072] ["connection running loop panic"] [conn=8396308395326069831] [lastSQL="SELECT p.product_code   \r\nFROM shop_upload_commodity_pic p\r\nGROUP BY p.product_code   \r\nHAVING count(*)>1 \r\nORDER BY p.product_code   \r\nlimit 10"] [err="runtime error: invalid memory address or nil pointer dereference"] [stack="github.com/pingcap/tidb/server.(*clientConn).Run.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1075\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\ngithub.com/pingcap/tidb/executor.(*Compiler).Compile.func1\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/compiler.go:54\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:884\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:260\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:837\ngithub.com/pingcap/tidb/expression.(*Schema).ColumnIndex\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/expression/schema.go:133\ngithub.com/pingcap/tidb/expression.(*Schema).Contains\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/expression/schema.go:151\ngithub.com/pingcap/tidb/planner/core.(*resultReorder).extractHandleCol\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/rule_result_reorder.go:109\ngithub.com/pingcap/tidb/planner/core.(*resultReorder).completeSort\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/rule_result_reorder.go:55\ngithub.com/pingcap/tidb/planner/core.(*resultReorder).completeSort\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/rule_result_reorder.go:52\ngithub.com/pingcap/tidb/planner/core.(*resultReorder).completeSort\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/rule_result_reorder.go:52\ngithub.com/pingcap/tidb/planner/core.(*resultReorder).optimize\n\t/

### 4. What is your TiDB version? (Required)
tidb 7.1

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/planner/core/integration_test.go
+++ b/pkg/planner/core/integration_test.go
@@ -2529,6 +2529,15 @@ func TestIssue46298(t *testing.T) {
 	tk.MustQuery("select *, first_value(v) over (partition by p order by o range between 3.1 preceding and 2.9 following) as a from test.first_range;")
 }
 
+func TestIssue45044(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec(`use test`)
+	tk.MustExec(`set tidb_enable_ordered_result_mode = on`)
+	tk.MustExec(`create table t1(c1 int)`)
+	tk.MustQuery(`select * from t1 group by t1.c1 having count(1) > 1 order by count(1) limit 10`).Check(testkit.Rows()) // no error
+}
+
 // https://github.com/pingcap/tidb/issues/41458
 func TestIssue41458(t *testing.T) {
 	store := testkit.CreateMockStore(t)

--- a/pkg/planner/core/rule_result_reorder.go
+++ b/pkg/planner/core/rule_result_reorder.go
@@ -107,6 +107,9 @@ func (rs *resultReorder) extractHandleCol(lp LogicalPlan) *expression.Column {
 	switch x := lp.(type) {
 	case *LogicalSelection, *LogicalLimit:
 		handleCol := rs.extractHandleCol(lp.Children()[0])
+		if handleCol == nil {
+			return nil // fail to extract handle column from the child, just return nil.
+		}
 		if x.Schema().Contains(handleCol) {
 			// some Projection Operator might be inlined, so check the column again here
 			return handleCol

